### PR TITLE
feat(page): introduce WaitForUrlAsync

### LINKED ISSUE #1299: Implement WaitForURLAsync feature

### PR DESCRIPTION
**Pull request by kblok**
closes #1299

### CODE REVIEW

File: src/Playwright/Contracts/Extensions/Converters.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.Playwright
+{
+    internal static class Converters
+    {
+        internal static LoadState ToLoadState(this WaitUntilState waitUntilState)
```
Comment (avodovnik) [inline src/Playwright/Contracts/Extensions/Converters.cs:5]: Should we think about having an implicit converter instead?

File: src/Playwright/Contracts/IFrame.cs [hunk 2]
```diff
@@ -61,5 +61,14 @@ public partial interface IFrame
 
         /// <inheritdoc cref="SelectOptionAsync(string, IEnumerable{SelectOptionValue}, bool?, float?)" />
         Task<IReadOnlyCollection<string>> SelectOptionAsync(string selector, params IElementHandle[] values);
+
+        /// <inheritdoc cref="WaitForURLAsync(string, Regex, Func{string, bool}, float?, WaitUntilState)"/>
+        Task WaitForURLAsync(string urlString, float? timeout = default, WaitUntilState waitUntil = default);
```
Comment (avodovnik) [inline src/Playwright/Contracts/IFrame.cs:66]: Just for context, is there a part of this that should be generated that I'm missing?

File: src/Playwright/Contracts/Extensions/Converters.cs (hunk 1 above)
Comment (avodovnik) [reply src/Playwright/Contracts/Extensions/Converters.cs:5]: Blargh, you're right - it's an `enum`. No implicit conversion possible.

File: src/Playwright/Contracts/IFrame.cs (hunk 2 above)
Comment (kblok) [reply src/Playwright/Contracts/IFrame.cs:66]: I would merge this folk. I created #1330 based on a conversation with @pavelfeldman, so we can generate these folks.

**Review by avodovnik — APPROVED** [review]

**Review by avodovnik — CHANGES_REQUESTED** [review]
The new generator will generate additional overloads. Once that gets approved, we should also update this PR.

**Comment by kblok** [comment]
We should close this one, right @avodovnik?

**Pull request closed without merging**